packages/ogmios/src/Ogmios/TxSubmissionClient.ts

```diff
@@ -0,0 +1,123 @@
+  context: InteractionContext;
+  evaluateTx: (bytes: string, additionalUtxoSet?: Utxo) => Promise<TxSubmission.EvaluationResult>;
+  submitTx: (bytes: string) => Promise<TxId>;
+  shutdown: () => Promise<void>;
+}
+
+const matchEvaluateTx = (data: string) => {
+  const response = safeJSON.parse(data) as Ogmios['EvaluateTxResponse'];
+  if ((response.type as string) !== 'jsonwsp/fault' && response.methodname !== 'EvaluateTx') {
+    return null;
+  }
+  return response;
+};
+
+/**
+ * Waits for a response with the reflection property requestId matching the one we send in the mirror
+ * property of the request.
+ *
+ * @param socket The wbesocket object.
+ * @param requestId The requestId, this will be used to filter our response.
+ * @param resolve The original promise resolve callback.
+ * @param reject The original promise reject callback.
+ */
+const waitForResponse =
+  (
+    socket: WebSocket,
+    requestId: string,
+    resolve: { (value: string | PromiseLike<string>): void; (arg0: string): void },
+    reject: { (reason?: never): void; (arg0: Error[]): void }
+  ) =>
+  (message: string) => {
+    const submitTxRes: Ogmios['SubmitTxResponse'] = safeJSON.parse(message);
+
+    if ((submitTxRes.type as string) !== 'jsonwsp/fault' && submitTxRes.methodname !== 'SubmitTx') {
+      return;
+    }
+
+    if (!submitTxRes.reflection || submitTxRes.reflection.requestId !== requestId) return;
+
+    // This is the response we are waiting for, we can unregister our callback from the socket.
+    socket.removeListener('message', waitForResponse(socket, requestId, resolve, reject));
+
+    const response = TxSubmission.handleSubmitTxResponse(submitTxRes);
+
+    if (TxSubmission.isTxId(response)) {
+      resolve(response);
+    } else {
+      reject(response);
+    }
+  };
+
+/**
+ * Create a client for submitting signed transactions to underlying Cardano chain.
+ */
+export const createTxSubmissionClient = async (context: InteractionContext): Promise<TxSubmissionClient> => {
+  const evaluateTxResponse = eventEmitterToGenerator(context.socket, 'message', matchEvaluateTx)() as AsyncGenerator<
+    Ogmios['EvaluateTxResponse']
+  >;
+
+  return Promise.resolve({
+    context,
+    evaluateTx: (bytes, additionalUtxoSet) => {
+      ensureSocketIsOpen(context.socket);
+      return send<TxSubmission.EvaluationResult>(async (socket) => {
+        socket.send(
+          safeJSON.stringify({
+            ...baseRequest,
+            args: {
+              ...(additionalUtxoSet !== undefined ? { additionalUtxoSet } : {}),
+              evaluate: bytes
+            },
+            methodname: 'EvaluateTx'
+          } as unknown as Ogmios['EvaluateTx'])
+        );
+
+        const response = TxSubmission.handleEvaluateTxResponse((await evaluateTxResponse.next()).value);
+
+        if (TxSubmission.isEvaluationResult(response)) {
+          return response;
+        }
+        throw response;
+      }, context);
+    },
```

Comment:
Let's just remove it until we need it since we're not implementing mirror/reflection for this right now and not using this function anywhere
```suggestion
```